marimo/_cli/sandbox.py

```diff
@@ -63,6 +74,9 @@ def _read_pyproject(script: str) -> Dict[str, Any] | None:
             line[2:] if line.startswith("# ") else line[1:]
             for line in matches[0].group("content").splitlines(keepends=True)
         )
+        # gracefully handle paths with '@ $path' to abspaths
+        # to meet pyproject expectation
+        content = re.sub(r"(@|--editable|-e)\s+(~)", expand_user_home, content)
```

Comment:
does `uv` not handle the home expansion?

Reply:
It's pyproject toml schema that doesn't allow home expansion. Try it on main and you'll see the path being rejected.